src/components/link/link.test.tsx

```diff
@@ -0,0 +1,569 @@
+// We've chosen to ignore this for the whole file as this warning fires 15 times.
+// It warns that `... anchors are primarily expected to navigate. Use the button element instead.`
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+
+import Link from "./link.component";
+import MenuContext from "../menu/menu.context";
+import { baseTheme } from "../../style/themes";
+
+describe("If `isSkipLink` provided", () => {
+  it("should render `Skip to main content` text inside of Link", () => {
+    render(
+      <Link href="#test" isSkipLink>
+        Test Content
+      </Link>
+    );
+
+    expect(screen.getByText("Skip to main content")).toBeInTheDocument();
+  });
+
+  it("should render correct styling", () => {
+    render(
+      <Link href="#test" isSkipLink>
+        Test Content
+      </Link>
+    );
+
+    const linkElement = screen.getByTestId("link-anchor");
+
+    expect(linkElement).toHaveStyle({
+      position: "absolute",
+      paddingLeft: "var(--spacing300)",
+      paddingRight: "var(--spacing300)",
+      lineHeight: "36px",
+      fontSize: "var(--fontSizes200)",
+      left: "-999em",
+      color: "var(--colorsUtilityYin090)",
+      zIndex: `${baseTheme.zIndex.aboveAll}`,
+      boxShadow: "inset 0 0 0 var(--spacing025) var(--colorsActionMajor500)",
+      border: "var(--spacing025) solid var(--colorsUtilityYang100)",
+    });
+  });
+});
+
+test("should not call the onClick function when `disabled` prop is true and clicked", async () => {
+  const spy = jest.fn();
+  render(
+    <Link disabled onClick={spy}>
+      Test Content
+    </Link>
+  );
+
+  const linkElement = screen.getByText("Test Content");
+  await userEvent.click(linkElement);
+
+  expect(spy).not.toHaveBeenCalled();
```

Comment:
**suggestion(non-blocking)**: I think you could also assert that it is disabled: `expect(linkElement).toBeDisabled()`